DeepSpeech.py

```diff
@@ -1042,11 +1044,13 @@ def train():
     # Possibly restore checkpoint
     start_epoch = 0
     if restore_checkpoint:
+        # Restore checkpoint created after epoch N, restart from epoch N+1
         checkpoint = tf.train.get_checkpoint_state(checkpoint_dir)
         if checkpoint and checkpoint.model_checkpoint_path:
             hibernation_path = checkpoint.model_checkpoint_path
-            start_epoch = int(checkpoint.model_checkpoint_path.split('-')[-1])
-            print 'Resuming training from epoch %d' % (start_epoch + 1)
+            last_epoch = int(checkpoint.model_checkpoint_path.split('-')[-1])
+            start_epoch = last_epoch+1
```

Comment:
Maybe I'm off-by one too. But I'm worried about what happens if we are doing 50 epochs
```python
epoch = 50
```
and we are restoring form the 49th checkpoint
```python
last_epoch = 49
start_epoch = 50
```
In this case the loop below
```python
for epoch in range(start_epoch, epochs):
    ...
```
does not run at all and the final lines of this `train()` method
```
...
    # Indicate optimization has concluded
    print "FINISHED Optimization",\
          "  Overall time:", format_duration(stopwatch(global_time)),\
          "  Training time:", format_duration(global_train_time)
    print

    return train_wer, dev_wer, hibernation_path
```
will return, as the for loop never ran,
```python
train_wer = 0.0
dev_wer = 0.0
```
which is wrong.